### PR TITLE
smartmontools.rb: update to the latestest (7.2) version

### PR DESCRIPTION
**Pull request by samm-git**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by yurikoles** [comment]
@samm-git 

according to https://www.smartmontools.org/browser/trunk/smartmontools/os_darwin.cpp#L9
```
SPDX-License-Identifier: GPL-2.0-or-later
```

File: Formula/smartmontools.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Smartmontools < Formula
   desc "SMART hard drive monitoring"
   homepage "https://www.smartmontools.org/"
-  url "https://downloads.sourceforge.net/project/smartmontools/smartmontools/7.1/smartmontools-7.1.tar.gz"
-  sha256 "3f734d2c99deb1e4af62b25d944c6252de70ca64d766c4c7294545a2e659b846"
+  url "https://downloads.sourceforge.net/project/smartmontools/smartmontools/7.2/smartmontools-7.2.tar.gz"
+  sha256 "5cd98a27e6393168bc6aaea070d9e1cd551b0f898c52f66b2ff2e5d274118cd6"
   license "GPL-2.0"
```
Comment (yurikoles) [inline Formula/smartmontools.rb:6]:
```
license "GPL-2.0-or-later"
```

**Review by yurikoles — CHANGES_REQUESTED** [review]

File: Formula/smartmontools.rb (hunk 1 above)
Comment (samm-git) [reply Formula/smartmontools.rb]: @yurikoles done, thank you!

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
Thanks, @samm-git, @yurikoles!

**Pull request closed without merging**